### PR TITLE
Re:  Support for cloud storage connectors #172

### PR DESCRIPTION
**Pull request by shreyp135**
#178 
Added cloud storage connectors(sync and async) for aws and gcp.
Added tests for aws and gcp.
Added required env variables in the env template file.
Made changes according to the tests.

### CODE REVIEW

**Comment by shreyp135** [comment]
Hi @KaranrajM I have created a new pull request after necessary changes, all tests are passing except the google metadata access in gcp test. I am guessing that is specific to my system & network config.
Please check once.

**Comment by KaranrajM** [comment]
Hi @shreyp135 there are two test cases (test_write_file, test_download_file_to_temp_storage) failing for me as well. Please fix them. Also remove unnecessary imports from all the files which will clean up the code. Have you tested these GCP and AWS code with their actual keys just to check whether the code is working?

**Comment by s-cellulo** [comment]
Hi @KaranrajM, 
I am stuck on an error in the gcp connector, my account is verified and billing is also linked to the project but it is still showing this errors.<br>
image

**Comment by shreyp135** [comment]
Hi @KaranrajM,
I am stuck on this error, although my account is verified and billing is linked to the project these errors are persistent. Could you please help me in this ?

image

**Comment by KaranrajM** [comment]
> Hi @KaranrajM, I am stuck on this error, although my account is verified and billing is linked to the project these errors are persistent. Could you please help me in this ?
> 
> image

Can you try mocking the calls and pass the test cases? Also please do the suggested other changes as well. Have you tested this packages with your actual GCP and AWS keys to test storage functionalities?

**Comment by shreyp135** [comment]
Hi @KaranrajM,
I finally got all the test passing but while testing with actual keys, there is an authentication issue but I am assuming that is specific to my billing account.

File: jb-lib/lib/file_storage/gcp/gcp_sync_storage.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        return tmp_file_path
+
+    def public_url(self, file_path: str) -> str:
+        if not self.__client__:
+            raise Exception("GCPSyncStorage client not initialized")
+
+        bucket = self.__client__.bucket(self.__bucket_name__)
+        blob = bucket.blob(file_path)
+
+        url = blob.generate_signed_url(expiration=timedelta(days=1))
```
Comment (KaranrajM) [inline jb-lib/lib/file_storage/gcp/gcp_sync_storage.py:68]: Missing import for timedelta

File: jb-lib/lib/file_storage/gcp/gcp_storage.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    def __init__(self):
+        logger.info("Initializing GCP Storage")
+        bucket_name = os.getenv("GCP_BUCKET_NAME")
+        if not bucket_name:
+            raise ValueError(
+                "GCPAsyncStorage client not initialized. Missing bucket_name "
+            )
+        self.__bucket_name__ = bucket_name
+        self.__client__ = Storage()
```
Comment (KaranrajM) [inline jb-lib/lib/file_storage/gcp/gcp_storage.py:23]: Please check this Storage() instance from google package when u try to run this, it throws an error. Also try all the functions with keys to confirm whether the packages are working or not.

**Comment by KaranrajM** [comment]
Hi @shreyp135 I have left some comments, I tried running the gcp files with my keys, they were throwing errors. Please check them from ur side and once every storage function in GCP and AWS is running and working, we can close this PR

**Comment by Lekhanrao** [comment]
@shreyp135, any updates here please? Also this PR is for Issue #172 and not 178!

**Comment by shreyp135** [comment]
> @shreyp135, any updates here please? Also this PR is for Issue #172 and not 178!

Hi @Lekhanrao,
Sorry but I am not able to debug the errors that are persistent, but I believe that others are also working on this issue so this PR can be a good head start.

**Comment by KaranrajM** [comment]
Closing this PR. This PR is used for reference for future works. Thanks @shreyp135 for all the works you did here.

**Pull request closed without merging**